hostinfo/packages.lua

```diff
@@ -14,64 +14,82 @@ See the License for the specific language governing permissions and
 limitations under the License.
 --]]
 local HostInfo = require('./base').HostInfo
+local misc = require('./misc')
+local getInfoByVendor = misc.getInfoByVendor
+local run = misc.run
+local Transform = require('stream').Transform
 
-local fmt = require('string').format
-local los = require('los')
-local sigar = require('sigar')
-local table = require('table')
-local execFileToBuffers = require('./misc').execFileToBuffers
+--------------------------------------------------------------------------------------------------------------------
+local Reader = Transform:extend()
 
---[[ Packages Variables ]]--
+function Reader:initialize()
+  Transform.initialize(self, {objectMode = true})
+end
+
+local LinuxReader = Reader:extend()
+function LinuxReader:_transform(line, cb)
+  line = line:gsub("^%s*(.-)%s*$", "%1")
+  local _, _, key, value = line:find("(.*)%s(.*)")
+  if key then self:push({ name = key, version = value }) end
+  cb()
+end
+
+local MacReader = Reader:extend()
+function MacReader:_transform(line, cb)
+  self:push({ name = line, version = 'unknown' })
+  cb()
+end
+
+--------------------------------------------------------------------------------------------------------------------
+--[[ Last logins ]]--
 local Info = HostInfo:extend()
 function Info:initialize()
   HostInfo.initialize(self)
 end
 
 function Info:_run(callback)
-  if los.type() ~= 'linux' then
-    self._error = 'Unsupported OS for Packages'
-    return callback()
-  end
+  local errTable, outTable = {}, {}
+  local deb, rhel
+  deb = {cmd = 'dpkg-query', args = {'-W'}}
```

Comment:
local deb... Let's keep the format:

deb = {
  cmd = 'dpkg-query',
  args = {'-W'}
}